src/include/detail/linalg/tdb_matrix.h

```diff
@@ -122,21 +142,39 @@ class tdbBlockedMatrix : public MatrixBase {
 
   /**
    * @brief Construct a new tdbBlockedMatrix object, limited to `upper_bound`
-   * vectors. In this case, the `Matrix` is row-major, so the number of vectors
-   * is the number of rows.
+   * vectors. We read rows from 0 -> row domain length and cols from 0 -> col
+   * domain length. In this case, the `Matrix` is column-major, so the number of
+   * vectors is the number of columns.
    *
    * @param ctx The TileDB context to use.
    * @param uri URI of the TileDB array to read.
+   * @param upper_bound The maximum number of vectors to read. Set to 0 for no
+   * upper bound.
+   * @param temporal_policy The TemporalPolicy to use for reading the array
+   * data.
    */
-  tdbBlockedMatrix(const tiledb::Context& ctx, const std::string& uri) noexcept
+  tdbBlockedMatrix(
```

Comment:
Per the extensive comment above, I think the correct interface is to have just a single constructor with `std::optional` for `start_row`, et al.  This will simplify things a lot (code-wise and conceptually) and also allow the array to just be opened once in order to access its schema (it is opened in the initializer of the general constructor) so that `begin_row`, `last_row`, `begin_col`, and `last_col` are cheaply obtained.

Reply:
Sure, updated to new approach.